Removed halt and replaced it with a delay and reboot

wel97459 opened the pull request:
Replaced the halt function with a delay and reboot, instead of an infinite loop. This removes an edge case were there's a bad brown out and the radio fails to initialize properly.

As @liamcottle suggested I have added an error message that is customizable to other failure cases. Available on node with a display enabled.

fschrempf commented:
> This could keep some boards from booting when there's dirty power, and the radio chip fails to init correctly.

And I guess that's exactly what is intended. If there is a critical failure, there is no reason to continue. Rebooting could cause nasty boot loops that won't help either.

wel97459 commented:
The halt function is the only called in one place is whole code base just for that radio init in the being of boot. It's definitely a contributing factor for solar nodes not coming back online after a dead battery state.

 I thinking add a message for node with screens is a good idea.

wel97459 commented:
I removed the halt function and added delayedReboot. The function takes in a message string and delay amount, and displays the message like in the image.

With no Display_Class defined it just delays and reboots.

![20251121_180750](https://github.com/user-attachments/assets/c3f270a4-b656-4052-8fe6-8b897538186a)
